backend/controller/account.controller.js

```diff
@@ -82,22 +82,22 @@ router.put("/update/:id", auth, async (req, res) => {
 router.delete("/delete/:id", auth, async (req, res) => {
   try {
     const id = req.params.id;
-
-    // find the account by id
-    const account = await Account.findByIdAndDelete(id);
-    const transactions = await Transaction.find({ accountId: id });
-    const user = await User.findById(req.userId);
-    user.set({ accounts: user.accounts.filter((acc) => acc != id) });
-    await user.save();
-
+    const account = await Account.findById(id).exec();
     let flag = false;
-
-    // if account exists, update the balance and name
-    if (transactions && account) {
-      transactions.forEach(async (transaction) => {
-        await Transaction.deleteOne({ _id: transaction._id });
+    if(account){
+      const user = await User.findById(req.userId);
+      const transactions = await Transaction.find({ accountId: account._id });
+      transactions?.forEach(async (t) => {
+        await Transaction.findByIdAndDelete(t._id);
       });
+      user.set({
```

Comment:
insteded of using user.findById and then apply changes
you can directly use updateOne or update many

eg: https://github.com/AniruddhaGawali/Spend-Wise/blob/master/backend/controller/transaction.controller.js

on line 82